### DISCUSSION TITLE
How to create and share remote branches?

### DISCUSSION THREAD

**Discussion by LyzV** [community/community#21693] [New to GitHub]
<p>How to create and share remote branches? I want to publish a local branch for future collaborative development. How to do it right? Several developers will insert changes into this branch. For exampe:</p>
<p>Host 1:</p>
<h1>git branch</h1>
<p> master<br>
*idea</p>
<h1>git push origin idea</h1>
<p>Host 2:</p>
<h1>git pull origin</h1>
<h1>git checkout originj/idea</h1>
<h1>git brunch idea</h1>
<h1>nano … make profit</h1>
<h1>git commit -a -m “profit”</h1>
<h1>git push origin idea</h1>
<p>But something is wrong. We can not post changes. I read the documentation, is there something about --set-upstream-to, but understood nothing :slight_smile:</p>

**Comment by gauseen** [comment]
<pre><code>first push origin you should ,

$ git push -u origin idea
</code></pre>

**Comment by LyzV** [comment, marked as answer]
<p>Thank you.</p>
<p>And another user use:</p>
<h1>git fetch origin</h1>
<h1>git checkout -b idea origin/idea</h1>
